remix.init/index.js

```diff
@@ -30,59 +84,57 @@ async function main({ rootDirectory, isTypeScript }) {
     // get rid of anything that's not allowed in an app name
     .replace(/[^a-zA-Z0-9-_]/g, "-");
 
-  const [appArc, env, packageJson, deployConfig, readme] = await Promise.all([
-    fs.readFile(APP_ARC_PATH, "utf-8"),
-    fs.readFile(EXAMPLE_ENV_PATH, "utf-8"),
-    fs.readFile(PACKAGE_JSON_PATH, "utf-8").then((s) => JSON.parse(s)),
-    fs.readFile(DEPLOY_YAML_PATH, "utf-8").then((s) => YAML.parse(s)),
-    fs.readFile(README_PATH, "utf-8"),
-  ]);
+  const [appArc, env, readme, deployWorkflow, vitestConfig, packageJson] =
+    await Promise.all([
+      fs.readFile(APP_ARC_PATH, "utf-8"),
+      fs.readFile(EXAMPLE_ENV_PATH, "utf-8"),
+      fs.readFile(README_PATH, "utf-8"),
+      readFileIfNotTypeScript(isTypeScript, DEPLOY_WORKFLOW_PATH, (s) =>
+        YAML.parse(s)
+      ),
+      readFileIfNotTypeScript(isTypeScript, VITEST_CONFIG_PATH),
+      PackageJson.load(rootDirectory),
+    ]);
 
   const newEnv = env.replace(
     /^SESSION_SECRET=.*$/m,
     `SESSION_SECRET="${getRandomString(16)}"`
   );
 
-  let saveDeploy = null;
-  if (!isTypeScript) {
-    delete packageJson.scripts.typecheck;
-    packageJson.scripts.validate = packageJson.scripts.validate.replace(
-      " typecheck",
-      ""
-    );
-
-    delete deployConfig.jobs.typecheck;
-    deployConfig.jobs.deploy.needs = deployConfig.jobs.deploy.needs.filter(
-      (n) => n !== "typecheck"
-    );
-    // only write the deploy config if it's changed
-    saveDeploy = fs.writeFile(DEPLOY_YAML_PATH, YAML.stringify(deployConfig));
-  }
+  updatePackageJson({ APP_NAME, isTypeScript, packageJson });
 
-  const newPackageJson =
-    JSON.stringify(sort({ ...packageJson, name: APP_NAME }), null, 2) + "\n";
-
-  await Promise.all([
+  const fileOperationPromises = [
     fs.writeFile(
       APP_ARC_PATH,
       appArc.replace("grunge-stack-template", APP_NAME)
     ),
     fs.writeFile(ENV_PATH, newEnv),
-    fs.writeFile(PACKAGE_JSON_PATH, newPackageJson),
-    saveDeploy,
     fs.writeFile(
       README_PATH,
       readme.replace(new RegExp("RemixGrungeStack", "g"), toLogicalID(APP_NAME))
     ),
+    packageJson.save(),
     fs.copyFile(
       path.join(rootDirectory, "remix.init", "gitignore"),
       path.join(rootDirectory, ".gitignore")
     ),
-    fs.rm(path.join(rootDirectory, ".github/ISSUE_TEMPLATE"), {
+    fs.rm(path.join(rootDirectory, ".github", "ISSUE_TEMPLATE"), {
       recursive: true,
     }),
-    fs.rm(path.join(rootDirectory, ".github/PULL_REQUEST_TEMPLATE.md")),
-  ]);
+    fs.rm(path.join(rootDirectory, ".github", "PULL_REQUEST_TEMPLATE.md")),
+  ];
+
+  if (!isTypeScript) {
+    fileOperationPromises.push(
+      ...cleanupDeployWorkflow(deployWorkflow, DEPLOY_WORKFLOW_PATH)
+    );
+
```

Comment:
In `remix.config.js`, change to `server: "./server.js",` and 
`path.relative(appDir, "cypress/support/test-routes/create-user.ts")`